pkg/operation/seed/seed.go

```diff
@@ -1660,3 +1664,44 @@ func cleanupOrphanExposureClassHandlerResources(ctx context.Context, c client.Cl
 
 	return nil
 }
+
+// resizeOrDeleteLokiDataVolumeIfStorageNotTheSame updates the garden Loki PVC if passed storage value is not the same as the current one.
+// Caution: If the passed storage capacity is less than the current one the existing PVC and its PV will be deleted.
+func resizeOrDeleteLokiDataVolumeIfStorageNotTheSame(ctx context.Context, k8sClient client.Client, newStorageQuantity resource.Quantity, log logrus.FieldLogger) error {
+	// Check if we need resizing
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := k8sClient.Get(ctx, kutil.Key(v1beta1constants.GardenNamespace, "loki-loki-0"), pvc); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	storageCmpResult := newStorageQuantity.Cmp(*pvc.Spec.Resources.Requests.Storage())
+	if storageCmpResult == 0 {
+		return nil
+	}
+
+	log.Infof("Scaling StatefulState %s/%s to zero in order to detach pvc %q", v1beta1constants.GardenNamespace, v1beta1constants.StatefulSetNameLoki, pvc.Name)
+	if err := kubernetes.ScaleStatefulSetAndWaitUntilIsScaled(ctx, k8sClient, kutil.Key(v1beta1constants.GardenNamespace, v1beta1constants.StatefulSetNameLoki), 0); client.IgnoreNotFound(err) != nil {
+		return err
+	}
+
+	switch {
+	case storageCmpResult > 0:
+		// json merge patch
+		patch := client.MergeFrom(pvc.DeepCopy())
+		pvc.Spec.Resources.Requests = corev1.ResourceList{
+			corev1.ResourceStorage: newStorageQuantity,
+		}
+		log.Infof("Patch garden/loki PVC %q to %q of storage", pvc.Name, newStorageQuantity.String())
+		if err := k8sClient.Patch(ctx, pvc, patch); err != nil {
+			log.Warnf("Patch garden/loki PVC %q to %q of storage fails: %v", pvc.Name, newStorageQuantity.String(), err)
```

Comment:
Why is the error not returned here?

Reply:
It will be good to not disturb the seed reconciliation with Loki PVC patch error. In this case, if there is an error only one Loki PVC will remain with the same size and we could fix it manually.